rpcs3/Emu/Io/usio.cpp

```diff
@@ -102,6 +105,46 @@ void usb_device_usio::control_transfer(u8 bmRequestType, u8 bRequest, u16 wValue
 
 extern bool is_input_allowed();
 
+void usb_device_usio::load_backup(const std::string& path)
+{
+	fs::file backup_file(path, fs::create + fs::read + fs::write + fs::lock);
+
+	usio_backup_file = std::move(backup_file);
+	if (!usio_backup_file)
+	{
+		usio_log.error("Failed to load the USIO Backup file: %s", path);
+		return;
+	}
+
+	std::vector<u8> buffer(0xE0);
+	if (usio_backup_file.size() != buffer.size())
+	{
+		usio_log.notice("Invalid USIO Backup file detected. Treating it as an empty file: %s", path);
+		return;
+	}
+
+	usio_backup_file.read(buffer.data(), buffer.size());
+	memcpy(g_fxo->get<usio_memory>().backup_memory.data(), buffer.data(), 0xB8);
+	memcpy(g_fxo->get<usio_memory>().last_game_status.data(), buffer.data() + 0xB8, 0x28);
+}
+
+void usb_device_usio::save_backup()
+{
+	if (!usio_backup_file)
+	{
+		return;
+	}
+
+	std::vector<u8> buffer(0xE0);
+	memcpy(buffer.data(), g_fxo->get<usio_memory>().backup_memory.data(), 0xB8);
+	memcpy(buffer.data() + 0xB8, g_fxo->get<usio_memory>().last_game_status.data(), 0x28);
+
+	usio_backup_file.seek(0, fs::seek_set);
+	usio_backup_file.trunc(buffer.size());
+	usio_backup_file.write(buffer.data(), buffer.size());
+	usio_backup_file.close();
```

Comment:
This line is not needed because it's part of fs::file dtor.

Reply:
Done.